pennylane/circuit_drawer/mpldrawer.py

```diff
@@ -0,0 +1,651 @@
+        figsize=None (Iterable): Allows users to manually specify the size of the figure.  Defaults
+            to scale with the size of the circuit via ``n_layers`` and ``n_wires``.
+
+    **Example**
+
+    .. code-block:: python
+
+        def example():
+
+            drawer = MPLDrawer(n_wires=5,n_layers=5)
+
+            drawer.label(["0","a",r"$|\Psi\rangle$",r"$|\theta\rangle$", "aux"])
+
+            drawer.box_gate(0, [0,1,2,3,4], "Entangling Layers", text_kwargs={'rotation':'vertical'})
+            drawer.box_gate(1, [0, 1], "U(θ)")
+
+            drawer.box_gate(1, 4, "Z")
+
+            drawer.SWAP(1, (2, 3))
+            drawer.CNOT(2, (0,2))
+
+            drawer.ctrl(3, [1,3], control_values = [True, False])
+            drawer.box_gate(3, 2, "H", zorder=2)
+
+            drawer.ctrl(4, [1,2])
+
+            drawer.measure(5, 0)
+
+            drawer.fig.suptitle('My Circuit', fontsize='xx-large')
+
+            return drawer
+
+        example()
+
+    .. figure:: ../../_static/drawer/example_basic.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+    .. UsageDetails::
+
+    This class uses matplotlib and pyplot.  The figure and axes objects can be accessed via ``drawer.fig``
+    and ``drawer.ax`` respectively for further configuration. For example, the above example circuit manipulates the
+    pyplot figure to set a title using ``drawer.fig.suptitle``. You can also save the images using ``plt.savefig``.
+
+    Each gate takes arguments in order of ``layer`` followed by ``wires``.  These translate to ``x`` and
+    ``y`` coordinates in the graph. Layer number (``x``) increases as you go right, and wire number
+    (``y``) increases as you go down.  The y-axis is inverted.  This also means you can pass non-integer values to either keyword.
+    For example, if you have a long label, the gate can span multiple layers and have extra width:
+
+    .. code-block:: python
+
+        drawer = MPLDrawer(2,2)
+        drawer.box_gate(layer=0, wires=1, text="X")
+        drawer.box_gate(layer=1, wires=1, text="Y")
+
+        # Gate between two layers
+        drawer.box_gate(layer=0.5, wires=0, text="Big Gate", extra_width=0.5)
+
+    .. figure:: ../../_static/drawer/float_layer.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+    You can globally control the style with ``plt.rcParams`` or
+    `styles <https://matplotlib.org/stable/tutorials/introductory/customizing.html>`_ .
+    If we customize ``plt.rcParams`` before executing our example function, we get a
+    different style:
+
+    .. code-block:: python
+
+        plt.rcParams['patch.facecolor'] = 'white'
+        plt.rcParams['patch.edgecolor'] = 'black'
+        plt.rcParams['patch.linewidth'] = 2
+        plt.rcParams['patch.force_edgecolor'] = True
+        plt.rcParams['lines.color'] = 'black'
+
+        example()
```

Comment:
Oh, I see now, so this would plot the example above again, but with the new colours/style.